File: src/main/java/net/usbwire/usbplus/util/chat/Clipboard.java
package net.usbwire.usbplus.util.chat;

import gg.essential.universal.wrappers.message.UMessage;
import gg.essential.universal.wrappers.message.UTextComponent;
import net.minecraft.text.ClickEvent;

public class Clipboard {
	public static UMessage clipboardBuilder(String name, String clipboard) {
		return clipboardBuilder(name, clipboard, new UMessage());
	}

	public static UMessage clipboardBuilder(String name, String clipboard, UMessage message) {
		// copy
		UTextComponent copyComponent = new UTextComponent(name);
		copyComponent.setClick(ClickEvent.Action.COPY_TO_CLIPBOARD, clipboard);
		message.addTextComponent(copyComponent);

		return message;
	}
}

File: src/main/kotlin/net/usbwire/usbplus/util/chat/Clipboard.kt
package net.usbwire.usbplus.util.chat

import gg.essential.universal.wrappers.message.*
import net.minecraft.text.*

object ChatUtil {
	fun clipboardBuilder(
		name: String,
		clipboard: String,
		message: UMessage = UMessage()
	): UMessage {
		// copy
		val copyComponent = UTextComponent("${name}")
		copyComponent.setClick(ClickEvent.Action.COPY_TO_CLIPBOARD, clipboard);
		message.addTextComponent(copyComponent)

		return message
	}
}